sfge/src/main/java/com/salesforce/graph/symbols/apex/ApexValueBuilder.java

```diff
@@ -435,9 +411,6 @@ public ApexSoqlValue buildSoql() {
     }
 
     public ApexStringValue buildString(String value) {
-        if (value == null) {
```

Comment:
I see no reason why UnexpectedException should ever be thrown.

We should simply return null (as the TODO above suggests).

In general, we need to revisit our null handling.

Reply:
This change confuses me. What's an example of a piece of code that wasn't evaluating before that evaluates after this change?

Reply:
Sure. I'll even add a test to show the following example:

String s = 'Foo';
System.debug(s.substringBetween('bar'));

WIthout removing this UnexepctedException, we would get the following stack trace:

ERROR ApexPathExpanderUtil:315 - Incomplete. Current ApexPathExpanders size=0
com.salesforce.exception.UnexpectedException: Use no-args version
	at com.salesforce.graph.symbols.apex.ApexValueBuilder.buildString(ApexValueBuilder.java:415) ~[main/:?]
	at com.salesforce.graph.symbols.apex.ApexStringValue.apply(ApexStringValue.java:548) ~[main/:?]

This makes no sense and this shouldn't crash. Removing this null check then gives me a value which passes the following conditions:

        ApexStringValue value = result.getVisitor().getSingletonResult();
        MatcherAssert.assertThat(value.isIndeterminant(), Matchers.equalTo(false));
        // Although determinant, the isPresent is false because the value is null
        MatcherAssert.assertThat(value.getValue().isPresent(), Matchers.equalTo(false));
        
I'll add a test with a new commit to show this.